### PR TITLE
DisplayApplicationDate

### PR DESCRIPTION
**Pull request by Mari-Hayashi**
Display the date on which the application is submitted on the list of applicants.
Student search remains unchanged.

search_applicants
search_student

### CODE REVIEW

File: src/client/src/Domains/Student/api/index.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export interface IStudentPreview {
     };
     department?: _IDepartment;
     classStanding?: classStandingTypes;
+    date?: Date;
```
Comment (johan1505) [inline src/client/src/Domains/Student/api/index.ts:53]: The date property of a job application is never undefined in the database, so this property shouldn't be optional

**Review by johan1505 — CHANGES_REQUESTED** [review]

File: src/client/src/Domains/Student/api/index.ts (hunk 1 above)
Comment (Mari-Hayashi) [reply src/client/src/Domains/Student/api/index.ts:53]: But I think the student search result is using the same component and they don't have date?

File: src/client/src/Components/CardPreview/index.tsx [hunk 2]
```diff
@@ -51,9 +51,11 @@ function CardPreview({
                     </Grid>
                     {Object.entries(values).map(([key, value]) => (
                         <Grid item xs={12}>
-                            <Typography variant='body1'>
-                                {key}: {value}
-                            </Typography>
+                            {value === undefined ? (<> </>) : (
```
Comment (johan1505) [inline src/client/src/Components/CardPreview/index.tsx]: Is it possible to just do:

```
{ value  &&  (
            <Typography variant='body1'>
                {key}: {value}
            </Typography>
    )}
```

File: src/client/src/Components/CardPreview/index.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ interface CardPreviewProps {
     onClick: () => void;
     isSelected: boolean;
     title: string;
-    values: { [key: string]: string | number | JSX.Element };
+    values: { [key: string]: string | number | JSX.Element | undefined };
```
Comment (johan1505) [inline src/client/src/Components/CardPreview/index.tsx:26]: Is it possible to do 

`    values: { [key: string]?: string | number | JSX.Element};
`

File: src/client/src/Domains/Student/api/index.ts (hunk 1 above)
Comment (johan1505) [reply src/client/src/Domains/Student/api/index.ts:53]: got it

**Review by johan1505 — CHANGES_REQUESTED** [review]
left comments

File: src/client/src/Components/CardPreview/index.tsx (hunk 2 above)
Comment (Mari-Hayashi) [reply src/client/src/Components/CardPreview/index.tsx]: Done.

File: src/client/src/Components/CardPreview/index.tsx (hunk 3 above)
Comment (Mari-Hayashi) [reply src/client/src/Components/CardPreview/index.tsx:26]: Not possible.

**Pull request merged**